.github/workflows/tests.yml

```
@@ -0,0 +1,88 @@
+name: Run metaflow_custom tests
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  run_test:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+    runs-on: ${{ matrix.os }}
+    name: Run pytest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install Mac dependencies
+        if: runner.os == 'macOS'
+        run: |
+          brew install git-crypt coreutils
+          $CONDA/bin/python -m pip install --upgrade pip
+          $CONDA/bin/python -m pip install -r requirements.txt
+          $CONDA/bin/python -m pip install .
+          conda config --add channels conda-forge
+
+      - name: Fix Conda permissions on macOS
+        if: runner.os == 'macOS'
+        run: sudo chown -R $UID $CONDA
+
+      - name: Install Linux dependencies
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y git-crypt
+          $CONDA/bin/python -m pip install --upgrade pip
+          $CONDA/bin/python -m pip install -r requirements.txt
+          $CONDA/bin/python -m pip install .
+          conda config --add channels conda-forge
```

Suggestion to factor out commonalities

```suggestion
      - name: Install Mac dependencies and permissions
        if: runner.os == 'macOS'
        run: |
          brew install git-crypt coreutils
          sudo chown -R $UID $CONDA

      - name: Install Linux dependencies
        if: runner.os == 'Linux'
        run: |
          sudo apt-get update -y && sudo apt-get install -y git-crypt

      - name: Install cross-platform dependencies
        run: |
          $CONDA/bin/python -m pip install --upgrade pip
          $CONDA/bin/python -m pip install -r requirements.txt
          $CONDA/bin/python -m pip install .
          conda config --add channels conda-forge
```